File: 2025/07/prog_84512_모음사전.js
/*
완탐 
모든 경우의 수 
*/
let answer = -1
function recur(alpha, sel, target, idx) {
    answer += 1
    
    // 성공 
    if (sel.join("") === target) {
        return true
    }
    
    // 실패 
    if (sel.length === 5) {
        return false
    }
    
    for (let i = 0; i < alpha.length; i++) {
        sel.push(alpha[i])
        if (recur(alpha, sel, target, i+1)) return true
        sel.pop()
    }
}



function solution(word) {
    const alpha = ['A', 'E', 'I', 'O', 'U']
    recur(alpha, [], word, 0)
    return answer;
}

File: 2026/01/prog_84512_모음사전.py
# 완탐
# 중복조합 
answer = -1
def solution(word):
    alpha = ['A', 'E', 'I', 'O', 'U']
    
    def recur(sel):
        global answer 
        answer += 1

        if sel == word:
            return True
        
        # 기저 조건 
        if len(sel) == 5:
            return False
        
        for i in range(len(alpha)):
            sel += alpha[i]
            if recur(sel): return True
            sel = sel[:-1]
            
    recur('')
    
    return answer